From 055e608c1b9db1110ad3ffd4bfa87ab9948e8e11 Mon Sep 17 00:00:00 2001
From: Shazad Hussain <shazhuss@qti.qualcomm.com>
Date: Mon, 22 Dec 2025 15:30:33 +0530
Subject: [PATCH 1/2] PENDING: qcom: Add sa7255p compatibles for core drivers

Add sa7255p specific compatible for uart, i2c, spi and ufs to
accomodate scmi changes.

Change-Id: Ia695287fba8201a09aa560be59a2dc389bf9e29a
Signed-off-by: Shazad Hussain <shazhuss@qti.qualcomm.com>
Signed-off-by: Sriram Dash <sriramd@qti.qualcomm.com>
---
 drivers/i2c/busses/i2c-qcom-geni.c    |  4 +++-
 drivers/soc/qcom/qcom-geni-se.c       |  1 +
 drivers/spi/spi-geni-qcom.c           |  4 +++-
 drivers/tty/serial/qcom_geni_serial.c | 12 +++++++++++-
 drivers/ufs/host/ufs-qcom.c           |  1 +
 5 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-qcom-geni.c b/drivers/i2c/busses/i2c-qcom-geni.c
index bd6d0c280634..d742cb214d45 100644
--- a/drivers/i2c/busses/i2c-qcom-geni.c
+++ b/drivers/i2c/busses/i2c-qcom-geni.c
@@ -1005,7 +1005,8 @@ static int geni_i2c_probe(struct platform_device *pdev)
 
 	gi2c->dev_data = desc;
 
-	if (!of_device_is_compatible(dev->of_node, "qcom,sa8255p-geni-i2c")) {
+	if (!(of_device_is_compatible(dev->of_node, "qcom,sa8255p-geni-i2c") ||
+			of_device_is_compatible(dev->of_node, "qcom,sa7255p-geni-i2c"))) {
 		if (desc->has_core_clk) {
 			gi2c->core_clk = devm_clk_get(dev, "core");
 			if (IS_ERR(gi2c->core_clk))
@@ -1226,6 +1227,7 @@ static const struct geni_i2c_desc i2c_master_hub = {
 static const struct of_device_id geni_i2c_dt_match[] = {
 	{ .compatible = "qcom,geni-i2c", .data = &geni_i2c },
 	{ .compatible = "qcom,geni-i2c-master-hub", .data = &i2c_master_hub },
+	{ .compatible = "qcom,sa7255p-geni-i2c", .data = &remotely_geni_i2c },
 	{ .compatible = "qcom,sa8255p-geni-i2c", .data = &remotely_geni_i2c },
 	{}
 };
diff --git a/drivers/soc/qcom/qcom-geni-se.c b/drivers/soc/qcom/qcom-geni-se.c
index fbd16a46c399..a5689e0079aa 100644
--- a/drivers/soc/qcom/qcom-geni-se.c
+++ b/drivers/soc/qcom/qcom-geni-se.c
@@ -1416,6 +1416,7 @@ static const struct geni_se_desc i2c_master_hub_desc = {
 static const struct of_device_id geni_se_dt_match[] = {
 	{ .compatible = "qcom,geni-se-qup", .data = &qup_desc },
 	{ .compatible = "qcom,geni-se-i2c-master-hub", .data = &i2c_master_hub_desc },
+	{ .compatible = "qcom,sa7255p-geni-se-qup", .data = &remotely_qup_desc },
 	{ .compatible = "qcom,sa8255p-geni-se-qup", .data = &remotely_qup_desc },
 	{}
 };
diff --git a/drivers/spi/spi-geni-qcom.c b/drivers/spi/spi-geni-qcom.c
index 723eb6d1bf5c..dc9c1a9f636e 100644
--- a/drivers/spi/spi-geni-qcom.c
+++ b/drivers/spi/spi-geni-qcom.c
@@ -1245,7 +1245,8 @@ static int spi_geni_probe(struct platform_device *pdev)
 	if (!mas->dev_data)
 		return -ENXIO;
 
-	if (of_device_is_compatible(dev->of_node, "qcom,sa8255p-geni-spi")) {
+	if (of_device_is_compatible(dev->of_node, "qcom,sa8255p-geni-spi") ||
+			of_device_is_compatible(dev->of_node, "qcom,sa7255p-geni-spi")) {
 		ret = dev_pm_domain_attach_list(&pdev->dev, &mas->dev_data->pd_data,
 						&mas->se.pd_list);
 		if (ret < 0)
@@ -1473,6 +1474,7 @@ static const struct geni_spi_desc remotely_geni_spi = {
 
 static const struct of_device_id spi_geni_dt_match[] = {
 	{ .compatible = "qcom,geni-spi", .data = &geni_spi },
+	{ .compatible = "qcom,sa7255p-geni-spi", .data = &remotely_geni_spi },
 	{ .compatible = "qcom,sa8255p-geni-spi", .data = &remotely_geni_spi },
 	{}
 };
diff --git a/drivers/tty/serial/qcom_geni_serial.c b/drivers/tty/serial/qcom_geni_serial.c
index 092b0b4898ca..b5fd88ee6260 100644
--- a/drivers/tty/serial/qcom_geni_serial.c
+++ b/drivers/tty/serial/qcom_geni_serial.c
@@ -1808,7 +1808,9 @@ static int qcom_geni_serial_probe(struct platform_device *pdev)
 	port->se.wrapper = dev_get_drvdata(pdev->dev.parent);
 
 	if (of_device_is_compatible(pdev->dev.of_node, "qcom,sa8255p-geni-debug-uart") ||
-	    of_device_is_compatible(pdev->dev.of_node, "qcom,sa8255p-geni-uart")) {
+	    of_device_is_compatible(pdev->dev.of_node, "qcom,sa8255p-geni-uart") ||
+		of_device_is_compatible(pdev->dev.of_node, "qcom,sa7255p-geni-debug-uart") ||
+		of_device_is_compatible(pdev->dev.of_node, "qcom,sa7255p-geni-uart")) {
 		ret = dev_pm_domain_attach_list(&pdev->dev,
 						&port->dev_data->pd_data, &port->se.pd_list);
 		if (ret < 0)
@@ -2041,6 +2043,10 @@ static const struct of_device_id qcom_geni_serial_match_table[] = {
 		.compatible = "qcom,geni-debug-uart",
 		.data = &qcom_geni_console_data,
 	},
+	{
+		.compatible = "qcom,sa7255p-geni-debug-uart",
+		.data = &remotely_qcom_geni_console_data,
+	},
 	{
 		.compatible = "qcom,sa8255p-geni-debug-uart",
 		.data = &remotely_qcom_geni_console_data,
@@ -2049,6 +2055,10 @@ static const struct of_device_id qcom_geni_serial_match_table[] = {
 		.compatible = "qcom,geni-uart",
 		.data = &qcom_geni_uart_data,
 	},
+	{
+		.compatible = "qcom,sa7255p-geni-uart",
+		.data = &remotely_qcom_geni_uart_data,
+	},
 	{
 		.compatible = "qcom,sa8255p-geni-uart",
 		.data = &remotely_qcom_geni_uart_data,
diff --git a/drivers/ufs/host/ufs-qcom.c b/drivers/ufs/host/ufs-qcom.c
index 9bd26c63476f..264b4612acd7 100644
--- a/drivers/ufs/host/ufs-qcom.c
+++ b/drivers/ufs/host/ufs-qcom.c
@@ -2415,6 +2415,7 @@ static void ufs_qcom_resume_complete(struct device *dev)
 
 static const struct of_device_id ufs_qcom_of_match[] __maybe_unused = {
 	{ .compatible = "qcom,ufshc"},
+	{ .compatible = "qcom,sa7255p-ufshc", .data = &ufs_hba_qcom_fmr_vops},
 	{ .compatible = "qcom,sa8255p-ufshc", .data = &ufs_hba_qcom_fmr_vops},
 	{},
 };
-- 
2.34.1

